### PR TITLE
[Net] Prevent un-necessary disconnections in version handshake

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Discard known stale protocol enforcement spork data during startup.

Quick workaround for clients that have not yet seen the new spork 15 broadcast since they last ran.

This evaluates the signing time of spork 15 (used to disconnect peers in the version handshake) and
discards the data of known-stale information. This will prevent the need for such clients to manually
delete the `sporks` folder in their datadir.

### CODE REVIEW

File: src/spork.cpp [hunk 1]
```diff
@@ -57,6 +57,20 @@ void CSporkManager::LoadSporksFromDB()
...
+        // TODO: Temporary workaround for v5.0 clients to ensure up-to-date protocol version spork
+        if (sporkDef.sporkId == SPORK_15_NEW_PROTOCOL_ENFORCEMENT_2) {
+            LogPrintf("%s : Spork 15 signed at %d\n", __func__, pSporkDB->GetSporkTime(SPORK_15_NEW_PROTOCOL_ENFORCEMENT_2));
+            // 1578338986 is the timestamp that spork 15 was last signed at for mainnet for the previous
+            // protocol bump. If the timestamp in the DB is equal or lower than this, we know that
+            // the value is stale and should ignore it to prevent un-necessary disconnections in the
+            // version handshake process. This value is also suitable for testnet as the timestamp
+            // for this spork on that network was signed shortly after this.
+            if (pSporkDB->GetSporkTime(SPORK_15_NEW_PROTOCOL_ENFORCEMENT_2) <= 1578338986 ) {
+                LogPrintf("%s : Stale spork 15 detected, ignoring...\n", __func__);
```
Comment (random-zebra) [inline src/spork.cpp:69]: Let's not make a second database read. Better to save the value in a local variable at line 62, and reuse it here.

File: src/spork.cpp [hunk 2]
```diff
@@ -57,6 +57,20 @@ void CSporkManager::LoadSporksFromDB()
...
+                continue;
```
Comment (random-zebra) [inline src/spork.cpp:70]: instead of `continue`, we should probably consider **all** spork values as stale, clear the spork manager, and return.

**Review by random-zebra — COMMENTED** [review]
Looking good. Just a couple suggestions.

File: src/spork.cpp (hunk 1 above)
Comment (furszy) [reply src/spork.cpp:69]: the value is already in a local variable, line 54. Can directly use it.

Comment (Fuzzbawls) [reply src/spork.cpp]: yep, updated. I originally had this check elsewhere but moved it here.

File: src/spork.cpp (hunk 2 above)
Comment (Fuzzbawls) [reply src/spork.cpp]: done

**Review by furszy — APPROVED** [review]
looking good now, utACK 9aae20b

**Review by random-zebra — APPROVED** [review]
utACK 9aae20b57f231aad6779c423d08dc859b0083442

**Pull request merged**